contracts/gas-service/AxelarGasService.sol

```diff
@@ -3,24 +3,36 @@
 pragma solidity ^0.8.0;
 
 import { IERC20 } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IERC20.sol';
+import { IAxelarGasService } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGasService.sol';
+import { InterchainGasEstimation } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/gas-estimation/InterchainGasEstimation.sol';
+import { Upgradable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/upgradable/Upgradable.sol';
 import { SafeTokenTransfer, SafeTokenTransferFrom } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/libs/SafeTransfer.sol';
 import { SafeNativeTransfer } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/libs/SafeNativeTransfer.sol';
-import { IAxelarGasService } from '../interfaces/IAxelarGasService.sol';
-import { Upgradable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/upgradable/Upgradable.sol';
 
 /**
  * @title AxelarGasService
  * @notice This contract manages gas payments and refunds for cross-chain communication on the Axelar network.
  * @dev The owner address of this contract should be the microservice that pays for gas.
  * @dev Users pay gas for cross-chain calls, and the gasCollector can collect accumulated fees and/or refund users if needed.
  */
-contract AxelarGasService is Upgradable, IAxelarGasService {
+contract AxelarGasService is InterchainGasEstimation, Upgradable, IAxelarGasService {
     using SafeTokenTransfer for IERC20;
     using SafeTokenTransferFrom for IERC20;
     using SafeNativeTransfer for address payable;
 
     address public immutable gasCollector;
 
+    enum GasPaymentType {
+        NativeForContractCall,
+        NativeForContractCallWithToken,
+        TokenForContractCall,
+        TokenForContractCallWithToken,
+        NativeForExpressCall,
+        NativeForExpressCallWithToken,
+        TokenForExpressCall,
+        TokenForExpressCallWithToken
+    }
```

Comment:
we don't want this logic atm, since we have fully scoped out requirements. `params` should be ignored.